message/name.go

```diff
@@ -70,7 +72,17 @@ func NameOf(m dogma.Message) Name {
 	}
 
 	rt := reflect.TypeOf(m)
-	n := typename.Of(rt)
+	n := goreflect.NameOf(rt)
+	return Name{n}
+}
+
+// NameOfType returns the fully-qualified type name of t.
+func NameOfType(t types.Type) Name {
```

Comment:
```suggestion
// NameFromType returns the fully-qualified type name of t.
func NameFromType(t types.Type) Name {
```

Suggesting this to bring naming in-line with  the `message package which has `message.TypeOf()` and `message.TypeFromReflect()`.